### PR TITLE
.NETCore & SQL Docker Image for Container Isolated Development & GitHub CodeSpaces

### PR DESCRIPTION
**Pull request by warrenbuckley**
## Resources
Below is a mix of documentation and resources that was used for this

TL:DR Using docker locally with WSL2 on Windows with the VSCode extension Remote Containers gives the same experience as GitHub CodeSpaces.

### VSCode Container Docs & tutorials
https://channel9.msdn.com/Series/Beginners-Series-to-Dev-Containers 📺 
https://channel9.msdn.com/Shows/Tabs-vs-Spaces/Visual-Studio-Code-Remote-dev-with-Containers--Codespaces 📺 
https://code.visualstudio.com/docs/remote/containers
https://docs.microsoft.com/en-us/learn/modules/use-docker-container-dev-env-vs-code/ 

## Docker Image & Software Installed
Docker Base Image from MS with SQL that allows .SQL scripts to be run on image creation
https://github.com/microsoft/vscode-dev-containers/tree/master/containers/dotnet-mssql
https://hub.docker.com/_/microsoft-vscode-devcontainers

* .NETCore 5.0
* NodeJS LTS installed
* SQL Server installed
* Blank DB created with .sql script called **UmbracoUnicore**
* SMTP4Dev

**SQL Server Login**
Server | Username | Password | Database Name
---|---|---|---
localhost | sa | P@ssw0rd | UmbracoUnicore

**SMTP 4 Dev**
This uses a docker image from [SMTP4Dev](https://github.com/rnwood/smtp4dev) a cross platform .NETCore SMTP server for testing
The UI to view emails sent from Umbraco such as invite flow emails or forgotten password emails etc can be viewed by visiting http://localhost:5000 or the magic port forwarded codespace equivalent. 

## Pre Installed VSCode Extensions in Container
To give a consistent experience for all users using this development container (local or GitHub CodeSpaces) the VSCode instance will have the following extensions installed

* C# (OmniSharp)
https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csharp

* SQL Server with a connection setup & preconfigured to the SQL Server Image 
https://marketplace.visualstudio.com/items?itemName=ms-mssql.mssql

## Using SQL Server Management Studio
If you wish to connect to the localhost SQL server running in the container locally you can connect with the same credentials but change the server name to `host.docker.internal`
Server | Username | Password | Database Name
---|---|---|---
host.docker.internal| sa | P@ssw0rd | UmbracoUnicore

## Related PR
The PR for automated user creation for unattended install from this PR has been merged into this PR
https://github.com/umbraco/Umbraco-CMS/pull/9934

### CODE REVIEW

**Comment by bergmania** [comment]
Seems to works as expected now :) 💪

**Review by bergmania — APPROVED** [review]
